src/main/groovy/life/qbic/samplestatus/reporter/Result.groovy

```diff
@@ -1,59 +1,54 @@
 package life.qbic.samplestatus.reporter
 
-import java.util.function.Supplier
-
 /**
  * <class short description - 1 Line!>
  *
  * <More detailed description - When to use, what it solves, etc.>
  *
  * @since <version tag>
  */
-class Result<S> implements Supplier<S>{
+class Result<S> {
 
     private final S data
     private final Exception exception
 
-    static Result<S> of(S value, Exception e) {
-        return new Result<S>(value, e)
+    static <S> Result<S> of(S value) {
+        return new Result(value)
+    }
+
+    static <S> Result<S> of(Exception e) {
+        return new Result(e)
     }
 
-    private Result(S data, Exception exception) {
+    private Result(S data) {
         this.data = data
+        this.exception = null
+    }
+
+    private Result(Exception exception) {
+        this.data = null
         this.exception = exception
     }
 
-    @Override
-    S get() {
-        if(!data) {
+    S getValue() throws NoSuchElementException {
+        if (!data) {
             throw new NoSuchElementException("Result with error has no value.")
         }
         return data
     }
 
-    Exception getError() {
-        if(!error) {
+    Exception getError() throws NoSuchElementException {
+        if (!error) {
             throw new NoSuchElementException("Result with value has no error.")
         }
         return error
     }
 
-    Boolean isError() {
+    Boolean hasError() {
         return exception
```

Comment:
`return exception as boolean` would be more readable as you can easily see that type conversion happens here.